runtime/src/bank.rs

```diff
@@ -2764,6 +2765,32 @@ impl Bank {
         }
     }
 
+    #[allow(dead_code)]
+    fn hash_rewards_into_partitions(
+        stake_rewards: &StakeRewards,
+        rewarded_epoch: Epoch,
+        rewards: u64,
+        num_partitions: usize,
+        thread_pool: &ThreadPool,
+    ) -> Vec<StakeRewards> {
+        let seed = rewarded_epoch ^ rewards;
+        let hasher = create_epoch_reward_hasher(seed);
+
+        let mut result = vec![vec![]; num_partitions];
+
+        thread_pool.install(|| {
+            let partitions = stake_rewards
+                .par_iter()
+                .map(|reward| address_to_partition(hasher, num_partitions, &reward.stake_pubkey))
+                .collect::<Vec<_>>();
```

Comment:
`collect()` is unnecessary?

Reply:
Without collect, we will need to add mutex to the vec of vec, which slows down the for loop below.

Reply:
or build the vec of vec after the threads have joined?  that should be fast anyway

Reply:
pushed a commit to avoid calling collect into vec. use DashMap instead. 
after thread runs ,then swap out the result.

```
hash: 2355us
copy: 45955us
```
```
hash+DashMap: 25565us
swap: 3us
```